lib/Lib.thy

```diff
@@ -2533,8 +2533,18 @@ next
     by (simp only: pv) (erule not_prefix_cases, auto intro: r1 r2 ih)
 qed
 
-lemma rsubst:
-  "\<lbrakk> P s; s = t \<rbrakk> \<Longrightarrow> P t"
+lemmas rsubst = back_subst[where a=s and b=t for s t]
+
+lemma rsubst2:
+  "\<lbrakk>P s x; s = t; x = y\<rbrakk> \<Longrightarrow> P t y"
+  by simp
+
+lemma rsubst3:
+  "\<lbrakk>P a b c ; a = s; b = t; c = u\<rbrakk> \<Longrightarrow> P s t u"
+  by simp
+
+lemma rsubst4:
+  "\<lbrakk>P a b c d; a = s; b = t; c = u; d = v\<rbrakk> \<Longrightarrow> P s t u v"
```

Comment:
It'd be good the get the naming consistent on these. I can see the reasoning for `rsubst3` and `rsubst4` with a, b for the first and s, t for the second list of arguments. We could do the same for `rsubst2`. Changing the old `rsubst` is probably a lost cause, and we do want similarity with `subst`. What do people think?

Reply:
I haven't looked much into how these are used for MCS, and I haven't given too much thought to this, but I should at least point out now that these rules (including the reformulation of `rsubst` into a `lemmas` statement) are copied straight from the `rt` branch

Reply:
Understood -- we should take the opportunity to make these better. The `rt` branch does not historically have very high library standards in terms of thought put into naming, styles, etc, because that wasn't a priority at the time. We do want to invest those kinds of considerations on `master`, though, because they will become ingrained when they land there.

Reply:
I think if we went for `a, b` and `s, t` for `rsubst2`, then that would be best

Reply:
Yup, happy with that.